docs/compatibility.md

```diff
@@ -323,17 +322,14 @@ Only a limited set of formats are supported when parsing dates.
 * `"MM/yyyy"`
```

Comment:
If we don't need a config for dates any more do we need this section of the documentation at all any more?

https://github.com/NVIDIA/spark-rapids/issues/869 was duplicated to https://github.com/NVIDIA/spark-rapids/issues/1091 so even if that is all that we keep here, it would be good to update the comment there.

Reply:
We still need this section until we also merge the timestamp PR (which will follow on from this one). Issue 869/1091 is now resolved and I have now added tests to confirm.